src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -408,6 +420,19 @@ int CNeuralNetwork::lbfgs_progress(void* instance,
 		int n, int k, int ls)
 {
 	SG_SINFO("Epoch %i: Error = %f\n",k, fx);
+
+	CNeuralNetwork* network = static_cast<CNeuralNetwork*>(instance);
+	SGVector<float64_t> gradients((float64_t*)g, network->get_num_parameters(), false);
+	for (int32_t i=0; i<network->m_num_layers; i++)
+	{
+		SGVector<float64_t> layer_gradients = network->get_section(gradients, i);
+		if (layer_gradients.vlen > 0)
+		{
+			SG_SINFO("Layer %i, Max Gradient: %g, Mean Gradient: %g.\n", i,
```

Comment:
Alright, i'll do it.
On 10 May 2015 23:10, "Sergey Lisitsyn" <notifications@github.com> wrote:

> In src/shogun/neuralnets/NeuralNetwork.cpp
> <https://github.com/shogun-toolbox/shogun/pull/2840#discussion_r30004276>:
>
> > @@ -408,6 +420,19 @@ int CNeuralNetwork::lbfgs_progress(void* instance,
> >  		int n, int k, int ls)
> >  {
> >  	SG_SINFO("Epoch %i: Error = %f\n",k, fx);
> > +
> > +	CNeuralNetwork* network = static_cast<CNeuralNetwork*>(instance);
> > +	SGVector<float64_t> gradients((float64_t*)g, network->get_num_parameters(), false);
> > +	for (int32_t i=0; i<network->m_num_layers; i++)
> > +	{
> > +		SGVector<float64_t> layer_gradients = network->get_section(gradients, i);
> > +		if (layer_gradients.vlen > 0)
> > +		{
> > +			SG_SINFO("Layer %i, Max Gradient: %g, Mean Gradient: %g.\n", i,
>
> For convenience, can we output the layer name as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/shogun-toolbox/shogun/pull/2840/files#r30004276>.
>